pennylane/compiler/compiler.py

```diff
@@ -0,0 +1,204 @@
+    compilers names and entry points.
+    """
+
+    reload(pkg_resources)
+    _refresh_compilers()
+
+
+def available_compilers() -> List[str]:
+    """Loads and returns a list of available compilers that are
+    installed and compatible with the :func:`~.qjit` decorator.
+
+    **Example**
+
+    This method returns the name of installed compiler packages supported in
+    PennyLane. For example, after installing the
+    `Catalyst <https://github.com/pennylaneai/catalyst>`__
+    compiler, this will now appear as an available compiler:
+
+    >>> qml.compiler.available_compilers()
+    ['catalyst']
+    """
+
+    # Reload installed packages and updates
+    # the class variable names_entrypoints
+    _reload_compilers()
+
+    return list(AvailableCompilers.names_entrypoints.keys())
+
+
+def available(compiler="catalyst") -> bool:
+    """Check the availability of the given compiler package.
+
+    Args:
+        compiler (str): name of the compiler package (default value is ``catalyst``)
+
+    Return:
+        bool: ``True`` if the compiler package is installed on the system
+
+    **Example**
+
+    Before installing the ``pennylane-catalyst`` package:
+
+    >>> qml.compiler.available("catalyst")
+    False
+
+    After installing the ``pennylane-catalyst`` package:
+
+    >>> qml.compiler.available("catalyst")
+    True
+    """
+
+    # It only refreshes the compilers names and entry points if the name
+    # is not already stored. This reduces the number of re-importing
+    # ``pkg_resources`` as it can be a very slow operation on systems
+    # with a large number of installed packages.
+
+    if compiler not in AvailableCompilers.names_entrypoints:
+        # Reload installed packages and updates
+        # the class variable names_entrypoints
+        _reload_compilers()
+
+    return compiler in AvailableCompilers.names_entrypoints
+
+
+def active_compiler() -> str:
+    """Check which compiler is activated inside a :func:`~.qjit` evaluation context.
+
+    This helper function may be used during implementation
+    to allow differing logic for circuits or operations that are
+    just-in-time compiled versus those that are not.
+
+    Return:
+        str or None: Name of the active compiler inside a :func:`~.qjit` evaluation context.
+
+    **Example**
+
+    For example, you can use this method in your hybrid program to execute it
+    conditionally whether is called inside :func:`~.qjit` with ``"catalyst"``
+    as the activate compiler or not.
+
+    .. code-block:: python
+
+        dev = qml.device("lightning.qubit", wires=2)
+
+        @qml.qnode(dev)
+        def circuit(phi, theta):
+            if qml.compiler.active() == "catalyst":
+                qml.RX(phi, wires=0)
+            qml.CNOT(wires=[0, 1])
+            qml.PhaseShift(theta, wires=0)
+            return qml.expval(qml.PauliZ(0))
+
+    >>> circuit(np.pi, np.pi / 2)
+    1.0
+    >>> qml.qjit(circuit)(np.pi, np.pi / 2)
+    -1.0
```

Comment:
nice example!

Reply:
I just did a bit of re-thinking about the support and realized this could have much simpler use cases than what I was arguing at first :upside_down_face: